eland/operations.py

```diff
@@ -527,6 +527,25 @@ def aggs(self, query_compiler, pd_aggs):
         results = self._metric_aggs(query_compiler, pd_aggs, numeric_only=False)
         return pd.DataFrame(results, index=pd_aggs)
 
+    def filter(self, query_compiler, items=None, like=None, regex=None):
+        # This function is only called for axis='index',
+        # DataFrame.filter(..., axis="columns") calls .drop()
+        if items is not None:
+            self.filter_index_values(
+                query_compiler, field=query_compiler.index.es_index_field, items=items
+            )
+            return
+        elif like is not None:
+            arg_name = "like"
+        else:
+            assert regex is not None
+            arg_name = "regex"
+
+        raise NotImplementedError(
```

Comment:
Note, this assumes that `_id` is the current index. If the index is changed to a time series or other, we may support this.